### PR TITLE
Extensions for GraphQLRequest

### LINKED ISSUE #352: Add Extensions to GraphQLRequest
Hello, currenly we implement [Apollo Persisted Query](https://www.apollographql.com/docs/apollo-server/performance/apq/) in custom GraphQLHttpMiddleware.
After split request deserialization in separate nuget it got a little harder to do)
Can we have extensions field in [GraphQLRequest](https://github.com/graphql-dotnet/server/blob/develop/src/Core/Common/GraphQLRequest.cs) like Inputs?

### PR DESCRIPTION
**Pull request by sungam3r**
fixes #352 

@benmccallum @SimonCropp @joemcbride What do you think? I see that work is underway on the first release of the http transport specification. Should we consider merge in the master branch of http spec as an indicator that we need to move implementation forward?

### CODE REVIEW

File: src/Core/Common/GraphQLRequest.cs [hunk 1]
```diff
@@ -5,9 +5,11 @@ public class GraphQLRequest
         public const string OPERATION_NAME_KEY = "operationName";
         public const string QUERY_KEY = "query";
         public const string VARIABLES_KEY = "variables";
+        public const string EXTENSIONS_KEY = "extensions";
 
         public string OperationName { get; set; }
         public string Query { get; set; }
         public Inputs Inputs { get; set; }
+        public Inputs Extensions { get; set; }
```
Comment (sungam3r) [inline src/Core/Common/GraphQLRequest.cs:13]: Using `Inputs` type here is a temporary measure. We can introduce a new type, although this one should fit.

**Comment by benmccallum** [comment]
Sorry @sungam3r , we've been moving to HotChocolate so my focus is really over there now.

**Comment by sungam3r** [comment]
@benmccallum It is a little sad for the development of this project. If this is not a secret, can you talk about the reasons?

**Comment by SimonCropp** [comment]
i think we need to ship a new stable asap

**Comment by sungam3r** [comment]
Well, it is a very good question. Relates to https://github.com/graphql-dotnet/graphql-dotnet/issues/1321

**Comment by benmccallum** [comment]
As I mentioned on another ticket, I'll give some proper feedback when I'm fully on HotChocolate and have had a moment to reflect, but as @SimonCropp alludes to, a real release cycle (including previews, rcs, blog posts, Slack group, requests for feedback) was definitely one of the biggest factors. I was always happy to use graphql-dotnet's previews, but at a certain point it's not enough as the project can't gain any momentum/clear direction without major/minor releases. There's nothing to gather around, plan around or get excited about. 

To be clear, I think you (and others) @sungam3r have done an absolutely amazing job to make real progress from where it was say a year ago! And with some release cycles and some community stuff to make it all more public, it'd continue to do really well. But HotChocolate is already there. The community is thriving and we're getting excited about things like v11 which has a new execution engine with almost zero allocations, (+ they've done metrics and it's numbers of times faster than graphql-dotnet) + it's gonna drop some real-time features like `@defer` and `@stream`. 

I'll list out some more things when I get some more time.

**Comment by benmccallum** [comment]
Essentially just biting the bullet and get a v3 out would be huge. I know you guys make breaking changes quite often, and that may be a reason for the v3 delay to finalise a bunch of breaking things into one, but frankly it'd be better to have v3, v4, v5 in quick succession, as it's much clearer than breaking changes between preview 1423 and 1424 lol. + it's much easier to document each time. + it shows progression.

I know this isn't any of your fault though, you're just doing the best you can with the access you've got!

**Comment by sungam3r** [comment]
@benmccallum We migrated to GitHub Actions CI and finally published v3.0.0 (and 3.1.0/3.1.1 already). See https://github.com/graphql-dotnet/graphql-dotnet/issues/1321#issuecomment-717656610

File: src/Core/Common/GraphQLRequest.cs (hunk 1 above)
Comment (sungam3r) [reply src/Core/Common/GraphQLRequest.cs:51]: I think to keep `Inputs` type for this property.

**Comment by sungam3r** [comment]
Also this PR fixes a lot of warnings from check annotations. See https://github.com/graphql-dotnet/server/pull/418/files

**Comment by sungam3r** [comment]
@Shane32 @joemcbride I want to merge and release ASAP after review.

**Comment by Shane32** [comment]
I’ll review today

**Review by Shane32 — APPROVED** [review]
As long as you’re sure we don’t need any of the items removed from gitignore, I’m good. What version number do you want to publish as?

**Comment by Shane32** [comment]
4.1.1 to mirror 3.1.1 release of GraphQL.NET?

**Comment by sungam3r** [comment]
There is new public property so 4.2.2
изображение

**Pull request merged**